pageserver/src/tenant.rs

```diff
@@ -3767,13 +3767,17 @@ async fn run_initdb(
         .spawn()?;
 
     tokio::select! {
-        initdb_output = initdb_command.wait_with_output() => {
-            let initdb_output = initdb_output?;
-            if !initdb_output.status.success() {
-                return Err(InitdbError::Failed(initdb_output.status, initdb_output.stderr));
+        exit_status = initdb_command.wait() => {
+            let exit_status = exit_status?;
+            if !exit_status.success() {
+                let mut stderr = initdb_command.stderr.take().unwrap();
+                let mut stderr_vec = Vec::new();
+                tokio::io::copy(&mut stderr, &mut stderr_vec).await?;
```

Comment:
I don't think this works as one might think: it only works if there is no stderr output or very little of it (less than the default kernel pipe buffer capacity). We need to collect the output like `initdb.output()?` did in another `tokio::select!` arm.